### PR TITLE
Add google map field to demo

### PR DESCRIPTION
**Pull request by pxpm**
added two google map field examples, one in `Owner` and other in `Passport` to show both saving options (same column json or multiple columns).

### CODE REVIEW

**Comment by EGYWEB-Mohamed** [comment]
Good Job @pxpm 🎉🎉

**Comment by jorgetwgroup** [comment]
Hello @pxpm 

Maps works great! i just have a fail with "Get my location button" (see screeshont)

screenshot-backpack test-2022 11 28-09_46_49

Screenshot 2022-11-28 at 09 46 09

But maps load normal and i can move pin and show me new location.

screenshot-backpack test-2022 11 28-09_45_38

Let me know if something is wrong with my config, i set my api key in "GOOGLE_PLACES_KEY".

Cheers.

**Comment by pxpm** [comment]
@jorgetwgroup you browser location is not available locally without https 👍 

@tabacitu can you create a subdomain for this branch online so that we fully test the location services ? Something like https://demomaps.backpackforlaravel.com ? 

Going through the route of getting https locally is a pain, trust me, I've been there more than once!

Cheers

**Comment by tabacitu** [comment]
Hmm... I can confirm they _work_ in both places but... I wouldn't agree that they _make sense_ in those CRUDs @pxpm . Quite the contrary - I think we had a few "clean" forms there with `Passport` and `Owner`, and adding this example there... bloats them up. I especially dislike how the Owner form looks now, with all that white space, not pretty at all:



Let's please:
- [ ] move the `google_map` field to the Monster form, at the bottom of the "Time and space" tab; that's where we have ALL our field types so... why wouldn't it be there?! plus, then it will be picked up by the other related CRUDs so we can easily test it's working with CrudField JS API, etc;
- [ ] if we need to show a different functionality for it (eg. store `lat` and `long` in different columns) we should also add it to another "bloated" CRUD, for example... Cave? Yeah, I think Cave is a good one, it only has two fields so it won't be messy at the end of that; plus, it makes more sense for a Cave to have `lat` and `long` than a Passport or Owner, right?

**Comment by pxpm** [comment]
Done both things, moved one of the map fields to monster that stores in a single fake column, and another one to cave that stores the inputs separatelly. 

👍

File: app/Http/Controllers/Admin/PetShop/OwnerCrudController.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ protected function setupCreateOperation()
     {
         CRUD::setValidation(OwnerRequest::class);
 
-        CRUD::field('name');
+        CRUD::field('name')->wrapper(['class' => 'form-group col-md-6']);
```
Comment (tabacitu) [inline app/Http/Controllers/Admin/PetShop/OwnerCrudController.php:65]: No need for this field to be smaller now that it doesn't have the google map. Also... `->size(6)` man... does the same thing... fluent syntax FTW! You're old-school, I know, I know 😅

```suggestion
        CRUD::field('name');
```

File: app/Http/Controllers/Admin/PetShop/PassportCrudController.php [hunk 2]
```diff
@@ -66,7 +66,7 @@ protected function setupCreateOperation()
     {
         CRUD::setValidation(PassportRequest::class);
 
-        CRUD::field('pet');
+        CRUD::field('pet')->wrapper(['class' => 'form-group col-md-6']);
```
Comment (tabacitu) [inline app/Http/Controllers/Admin/PetShop/PassportCrudController.php:69]:
```suggestion
        CRUD::field('pet');
```

**Review by tabacitu — CHANGES_REQUESTED** [review]

File: app/Models/PetShop/Owner.php [hunk 3]
```diff
@@ -16,7 +16,7 @@ class Owner extends Model
      * @var array
      */
     protected $fillable = [
-        'name',
+        'name', 'address',
```
Comment (tabacitu) [inline app/Models/PetShop/Owner.php:19]:
```suggestion
        'name',
```

File: app/Models/PetShop/Passport.php [hunk 4]
```diff
@@ -48,4 +52,20 @@ public function pet()
...
+                $location = json_decode($value);
+
+                return [
+                    'lat'          => $location->lat ?? '',
+                    'lng'          => $location->lng ?? '',
+                    'full_address' => $location->formatted_address ?? '',
+                ];
+            }
+        );
+    }
```
Comment (tabacitu) [inline app/Models/PetShop/Passport.php:70]:
```suggestion

```

File: database/migrations/2022_01_17_083405_create_owners_table.php [hunk 5]
```diff
@@ -16,6 +16,7 @@ public function up()
         Schema::create('owners', function (Blueprint $table) {
             $table->id();
             $table->string('name');
+            $table->text('address')->nullable();
```
Comment (tabacitu) [inline database/migrations/2022_01_17_083405_create_owners_table.php:19]:
```suggestion
```

File: app/Models/PetShop/Passport.php [hunk 6]
```diff
@@ -29,6 +30,9 @@ class Passport extends Model
         'colour',
         'notes',
         'country',
+        'lat',
+        'lng',
+        'location',
```
Comment (tabacitu) [inline app/Models/PetShop/Passport.php:35]:
```suggestion
```

File: app/Models/PetShop/Passport.php [hunk 7]
```diff
@@ -2,6 +2,7 @@
 
 namespace App\Models\PetShop;
 
+use Illuminate\Database\Eloquent\Casts\Attribute;
```
Comment (tabacitu) [inline app/Models/PetShop/Passport.php:5]:
```suggestion
```

Comment (tabacitu) [inline app/Models/PetShop/Passport.php:5]:
```suggestion
```

**Review by tabacitu — CHANGES_REQUESTED** [review]

**Review by tabacitu — APPROVED** [review]

**Pull request merged**